packages/jest-config/src/index.js

```diff
@@ -17,24 +17,34 @@ import readConfigFileAndSetRootDir from './read_config_file_and_set_root_dir';
 
 function readConfig(
   argv: Argv,
-  packageRoot: string,
+  packageRootOrConfig: Path | ProjectConfig,
   // Whether it needs to look into `--config` arg passed to CLI.
   // It only used to read initial config. If the initial config contains
   // `project` property, we don't want to read `--config` value and rather
   // read individual configs for every project.
   skipArgvConfigOption?: boolean,
+  parentConfigPath?: ?Path,
 ): {
   configPath: ?Path,
   globalConfig: GlobalConfig,
   hasDeprecationWarnings: boolean,
   projectConfig: ProjectConfig,
 } {
   let rawOptions;
-  let configPath;
+  let configPath = null;
 
-  // A JSON string was passed to `--config` argument and we can parse it
-  // and use as is.
-  if (isJSONString(argv.config)) {
+  if (typeof packageRootOrConfig !== 'string') {
+    if (parentConfigPath) {
+      rawOptions = packageRootOrConfig;
+      rawOptions.rootDir = parentConfigPath;
+    } else {
+      throw new Error(
+        'jest: Cannot use configuration as an object without a file path',
```

Comment:
Can you end this sentence with a full stop (.) ?

Also, please capitalize Jest.